### PR TITLE
Setup linter for ClojureScript

### PR DESCRIPTION
**Pull request by FrostyX**
I don't have any experience with ClojureScript linters yet, `M-x indent-region` was good enough so far. But I want to see how they work.

### CODE REVIEW

**Comment by FrostyX** [comment]
Turns out my Emacs config doesn't adhere to the standard indentation as well :-)

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -12,3 +12,15 @@ jobs:
...
+        uses: DeLaGuardo/setup-clojure@12.3
+        with:
+          clj-kondo: 2022.05.31
+          cljfmt: 0.10.2
+
+      - name: Run clj-kondo
+        run: clj-kondo --lint frontend/src
+
+      - name: Run cljfmt
+        run: cljfmt check frontend/src
```
Comment (nikromen) [inline .github/workflows/pre-commit.yml:26]: I can see that cljs-kondo has support for pre-commit hooks. Could you please add hook in https://github.com/clj-kondo/clj-kondo/blob/master/doc/ci-integration.md#pre-commit-framework to `.pre-commit-config.yaml` so we use only pre-commit hooks?

Comment (nikromen) [reply .github/workflows/pre-commit.yml:26]: seems like running it as container is working locally without having clj-kondo installed on my machine

```
  - repo: https://github.com/clj-kondo/clj-kondo
    rev: v2023.12.15
    hooks:
      - id: clj-kondo-docker
```

hook passes on my machine

Comment (FrostyX) [reply .github/workflows/pre-commit.yml:26]: @nikromen can you please let me know what is the motivation for using the hooks as suggested? I am a bit lost in the nuances of GitHub Actions.

Comment (nikromen) [reply .github/workflows/pre-commit.yml:26]:
```
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 96d1a53..9d3582a 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -42,3 +42,8 @@ repos:
         additional_dependencies:
           - types-PyYAML
           - types-requests
+
+  - repo: https://github.com/clj-kondo/clj-kondo
+    rev: v2023.12.15
+    hooks:
+      - id: clj-kondo-docker
```

**Comment by FrostyX** [comment]
I updated the PR to use `.pre-commit-config.yaml`. The output of the GitHub checks looks a bit worse than before but I understand the appeal of using it this way.

Works for me.

**Review by nikromen — APPROVED** [review]

**Pull request merged**